### PR TITLE
MGDOBR-146 Add GitHub actions script to run Shard operator integrat…

### PR DESCRIPTION
**Pull request by sutaakar**
…ion tests

GitHub actions script starts KinD cluster, installs Ingress, builds and installs Shard operator.
Check is triggered for any change in `shard-operator` module.

Signed-off-by: Karel Suta <ksuta@redhat.com>

[MGDOBR-146](https://issues.redhat.com/browse/MGDOBR-146) 

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by sutaakar** [comment]
Currently blocked by https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/113
Right now it will check that operator successfully starts, once [MGDOBR-147](https://issues.redhat.com/browse/MGDOBR-147) is implemented then PR check should be adjusted to run those tests too.

File: .github/workflows/Operator-CI.yaml
Comment (r00ta) [inline .github/workflows/Operator-CI.yaml:52]: If I'm not wrong once https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/113 is merged you don't need anymore to specify `-Devent-bridge.k8s.platform=k8s`. But please double check :)

File: .github/workflows/Operator-CI.yaml
Comment (r00ta) [inline .github/workflows/Operator-CI.yaml:57]: In the mvn clean install step you could specify `-Dquarkus.container-image.tag=0.0.1` and remove this `docker tag` command. But up to you, no strong opinion from me

File: .github/workflows/Operator-CI.yaml
Comment (r00ta) [inline .github/workflows/Operator-CI.yaml:63]: Same as before related to the `-Dquarkus.container-image.tag=0.0.1` comment

**Review by r00ta — APPROVED** [review]
Looks great!

File: .github/workflows/Operator-CI.yaml
Comment (sutaakar) [reply .github/workflows/Operator-CI.yaml:52]: I think so, will try once it is merged.

File: .github/workflows/Operator-CI.yaml
Comment (robpblake) [reply .github/workflows/Operator-CI.yaml:57]: @r00ta @sutaakar Is the purpose of the image tag clearer if its tagged at the GIT_SHA of the PR build? 

```
-Dquarkus.container-image.tag=$GITHUB_SHA
```
Took me a little while to realise that `0.0.1` is meaningless and just prevents a pull for `:latest`.

Comment (sutaakar) [reply .github/workflows/Operator-CI.yaml:57]: Fine for me.
I will adjust the comment to be more descriptive.

**Comment by sutaakar** [comment]
@r00ta @robpblake Ready for final check

File: .github/workflows/Operator-CI.yaml
Comment (danielezonca) [inline .github/workflows/Operator-CI.yaml:42]: What about using a tagged version instead of `main`? In this way we avoid that every build can take a different version.
Latest release seems to be [v1.0.4](https://github.com/kubernetes/ingress-nginx/releases/tag/controller-v1.0.4)
```suggestion
          kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v1.0.4/deploy/static/provider/kind/deploy.yaml
```

File: .github/workflows/Operator-CI.yaml
Comment (danielezonca) [inline .github/workflows/Operator-CI.yaml:5]: For now it is fine because it is a CI specific for the operator but I expect we will add also some end2end smoke test too

File: .github/workflows/Operator-CI.yaml
Comment (danielezonca) [inline .github/workflows/Operator-CI.yaml:29]: There are overlaps between this build and the other existing [CI.yaml file](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/.github/workflows/CI.yaml#L31-L42).
Maven version for example here is not specified while in the other job it is so we can centralise this handling.
I think we can (with another ticket) split them in reusable composite actions similar like [this](https://github.com/kiegroup/kogito-pipelines/blob/main/.ci/actions/maven/action.yml) and the [corresponding usage](https://github.com/kiegroup/kogito-runtimes/blob/main/.github/workflows/kogito-runtimes-pr.yml#L36-L41).
Wdyt?
/cc @r00ta @robpblake

**Review by robpblake — APPROVED** [review]

File: .github/workflows/Operator-CI.yaml
Comment (sutaakar) [reply .github/workflows/Operator-CI.yaml:29]: Sounds good to me

**Review by danielezonca — APPROVED** [review]

File: .github/workflows/Operator-CI.yaml
Comment (sutaakar) [reply .github/workflows/Operator-CI.yaml:29]: Reported as [MGDOBR-153](https://issues.redhat.com/browse/MGDOBR-153)

**Pull request merged**
